### PR TITLE
ani-cli: 4.8 -> 4.9

### PR DESCRIPTION
**Pull request by Rishik-Y**
# Adding support for 24.05

**Tests**: Ran it multiple times and is already present in NisOS 24.11

## Package Information

- Package name: ani-cli
- Latest released version: 4.9 (for 24.05)
- Current version on the unstable channel: 4.9
- Current version on the stable/release channel: 4.8

Changes: Minor changes so Latest version works in 24.05 stable

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [X] (Module updates) Added a release notes entry if the change is significant
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Maintainer ping:
@skykanin

### CODE REVIEW

**Review by skykanin — APPROVED** [review]

**Review by skykanin — APPROVED** [review]

File: pkgs/applications/video/ani-cli/default.nix [hunk 1]
```diff
@@ -1,55 +1,45 @@
...
+, catt
+, syncplay
+, ffmpeg
+, fzf
+, aria2
+, withMpv ? true, mpv
+, withVlc ? false, vlc
+, withIina ? false, iina
+, chromecastSupport ? false
+, syncSupport ? false
```
Comment (donovanglover) [inline pkgs/applications/video/ani-cli/default.nix:17]: Please remove the unnecessary formatting changes

File: pkgs/applications/video/ani-cli/default.nix [hunk 2]
```diff
@@ -1,55 +1,45 @@
...
-    ++ lib.optional chromecastSupport catt
-    ++ lib.optional syncSupport syncplay;
+    let player = []
+        ++ lib.optional withMpv mpv
+        ++ lib.optional withVlc vlc
+        ++ lib.optional withIina iina;
+    in [ gnugrep gnused curl fzf ffmpeg aria2 ]
+      ++ player
+      ++ lib.optional chromecastSupport catt
+      ++ lib.optional syncSupport syncplay;
```
Comment (donovanglover) [inline pkgs/applications/video/ani-cli/default.nix:42]: If a formatter is used, make sure that you're using nixfmt-rfc-style

File: pkgs/applications/video/ani-cli/default.nix [hunk 3]
```diff
@@ -64,7 +54,7 @@ stdenvNoCC.mkDerivation rec {
 
   meta = with lib; {
     homepage = "https://github.com/pystardust/ani-cli";
-    description = "A cli tool to browse and play anime";
+    description = "Cli tool to browse and play anime";
```
Comment (donovanglover) [inline pkgs/applications/video/ani-cli/default.nix:57]:
```suggestion
    description = "CLI tool to browse and play anime";
```

**Review by donovanglover — CHANGES_REQUESTED** [review]
Commit message should be `ani-cli: 4.8 -> 4.9`

File: pkgs/applications/video/ani-cli/default.nix (hunk 2 above)
Comment (Rishik-Y) [reply pkgs/applications/video/ani-cli/default.nix]: Okay will do that in future! 👍️

**Comment by Rishik-Y** [comment]
I Have made the necessary changes!
Could you please take a look at it now?

**Review by donovanglover — COMMENTED** [review]
24.05 is deprecated and ani-cli is already 4.9 in 24.11, so I think this PR should be closed

**Comment by Rishik-Y** [comment]
> 24.05 is deprecated and ani-cli is already 4.9 in 24.11, so I think this PR should be closed

Oh...
I was under the assumption that 24.05 was not depreciated
I am somewhat of a newbie here, so didn't wanna do changes in Master/unstable yet,
so to learn first i wanted to try in other stable version and i did get the response on not doing PR in 23.11 and older.
If what you said is true, please go ahead and close the PR.

**Comment by donovanglover** [comment]
You can see the deprecated label on https://search.nixos.org/ and https://status.nixos.org/

For contributing to nixpkgs, it's ideal to target the `master` branch and use backport labels as appropriate.
